app/apps/app.ts

```diff
@@ -0,0 +1,162 @@
+    this.intents = appConfig.intents || [];
+    this.manifestType = appConfig.manifestType;
+    this.manifest = appConfig.manifest;
+  }
+
+  public async start() {
+    logStream.next([`Starting app "${this.title || this.name}".`, 'info', 'Toolbar']);
+
+    let manifestJSON: any; // The manifestType determines the shape of the parsed manifest.
+    let errorMsg: string;
+
+    try {
+      manifestJSON = JSON.parse(this.manifest);
+    } catch (error) {
+      errorMsg = `Failed to parse "${this.title || this.name}"'s manifest.`;
+
+      logStream.next([errorMsg, 'error', 'Toolbar']);
+      throw new Error(errorMsg);
+    }
+
+    /* The toolbar is currently aware of the following manifest types:
+       'org.finos.fdc3.demo' - covers browser/exe and therefore is platform/host independent
+       'org.finos.fdc3.demo.host' - covers platform/host specific applications
+    */
+    switch (this.manifestType) {
+      case 'org.finos.fdc3.demo':
+
+        this.handleOrgFinosFdc3Demo(manifestJSON);
+        break;
+
+      default:
+
+        await this.handleOrgFinosFdc3DemoHost(manifestJSON);
+        break;
+    }
+  }
+
+  private handleOrgFinosFdc3Demo(manifestJSON: any) {
+    let errorMsg: string;
+
+    switch (manifestJSON.type) {
+      case 'browser':
+        const appUrl = manifestJSON.url;
+
+        if (typeof appUrl === 'undefined') {
+          errorMsg = `Browser application "${this.title || this.name}" is missing url field inside the manifest.`;
+
+          logStream.next([errorMsg, 'error', 'Toolbar']);
+          throw new Error(errorMsg);
+        }
+
+        shell.openExternal(appUrl);
+        break;
+      case 'exe':
+        const exePath = manifestJSON.exePath;
+
+        if (typeof exePath === 'undefined') {
+          errorMsg = `Exe application "${this.title || this.name}" is missing exePath field inside the manifest.`;
+
+          logStream.next([errorMsg, 'error', 'Toolbar']);
+          throw new Error(errorMsg);
+        }
+
+        execFile(exePath);
+        break;
+      default:
+        errorMsg = `Unknown type "${manifestJSON.type}" application "${this.title || this.name}" with manifest type "org.finos.fdc3.demo".`;
+
+        logStream.next([errorMsg, 'error', 'Toolbar']);
+        throw new Error(errorMsg);
+    }
+  }
+
+  private async handleOrgFinosFdc3DemoHost(manifestJSON: any) {
+    let errorMsg: string;
+
+    const fdc3Impl = (global as any).fdc3Impl;
+
+    if (!fdc3Impl) {
+      errorMsg = `No fdc3Impl implementation provided. Application "${this.title || this.name}" with manifest type "org.finos.fdc3.demo.host" can't be started.`;
+
+      logStream.next([errorMsg, 'error', 'Toolbar']);
+      throw new Error(errorMsg);
+    }
+
+    switch (manifestJSON.type) {
+      case 'host':
+        const wait = (ms: number, msg?: string) => new Promise((_, reject) => setTimeout(() => reject(msg), ms));
+
+        try {
+          await Promise.race([fdc3Impl.open(this.name), wait(3000)]);
```

Comment:
Timeout is removed since all the logic related to opening a “host” application is removed - Toolbar 1.0 doesn't support opening of "host" applications.